x-pack/plugins/observability_solution/infra/server/routes/entities/get_latest_entity.ts

```diff
@@ -27,17 +27,23 @@ export async function getLatestEntity({
   inventoryEsClient,
   entityId,
   entityType,
-  entityDefinitions,
+  entityManagerClient,
 }: {
   inventoryEsClient: ObservabilityElasticsearchClient;
   entityType: 'host' | 'container';
   entityId: string;
-  entityDefinitions: EntityDefinition[] | EntityDefinitionWithState[];
-}) {
-  const hostOrContainerIdentityField = entityDefinitions[0]?.identityFields?.[0]?.field;
+  entityManagerClient: EntityClient;
+}): Promise<Entity | undefined> {
+  const { definitions } = await entityManagerClient.getEntityDefinitions({
+    builtIn: true,
+    type: entityType,
+  });
+
+  const hostOrContainerIdentityField = definitions[0]?.identityFields?.[0]?.field;
   if (hostOrContainerIdentityField === undefined) {
-    return;
+    return { [SOURCE_DATA_STREAM_TYPE]: [] };
   }
+
   const latestEntitiesEsqlResponse = await inventoryEsClient.esql('get_latest_entities', {
     query: `FROM ${ENTITIES_LATEST_ALIAS}
```

Comment:
Not part of this PR, but I'd suggest using `params here instead of concatenating strings

```ts
 const latestEntitiesEsqlResponse = await inventoryEsClient.esql('get_latest_entities', {
    query: `FROM ${ENTITIES_LATEST_ALIAS}
        | WHERE ${ENTITY_TYPE} == ?
        | WHERE ${hostOrContainerIdentityField} == ?
        | KEEP ${SOURCE_DATA_STREAM_TYPE}
      `,
    params: [entityType, entityId],
  });
```